dppl/_memory.pyx

```diff
@@ -0,0 +1,107 @@
+import dppl
+from dppl.backend cimport *
+from ._sycl_core cimport SyclContext, SyclQueue
+
+from cpython cimport Py_buffer
+
+
+cdef class Memory:
+    cdef DPPLSyclUSMRef memory_ptr
+    cdef Py_ssize_t nbytes
+    cdef SyclQueue queue
+
+    cdef _cinit(self, Py_ssize_t nbytes, ptr_type):
+        cdef SyclQueue q
+        cdef DPPLSyclUSMRef p
+
+        self.memory_ptr = NULL
+        self.queue = None
+        self.nbytes = 0
+
+        if (nbytes > 0):
+            q = dppl.get_current_queue()
+
+            if (ptr_type == "shared"):
+                p = DPPLmalloc_shared(nbytes, q.get_queue_ref())
+            elif (ptr_type == "host"):
+                p = DPPLmalloc_host(nbytes, q.get_queue_ref())
+            elif (ptr_type == "device"):
+                p = DPPLmalloc_device(nbytes, q.get_queue_ref())
+            else:
+                raise RuntimeError("Pointer type is unknown: {}" \
+                    .format(ptr_type))
+
+            if (p):
+                self.memory_ptr = p
+                self.nbytes = nbytes
+                self.queue = q
+            else:
+                raise RuntimeError("Null memory pointer returned")
+        else:
+            raise ValueError("Non-positive number of bytes found.")
+
+    def __dealloc__(self):
+        if (self.memory_ptr):
+            DPPLfree(self.memory_ptr, self.queue.get_queue_ref())
+        self.memory_ptr = NULL
+        self.nbytes = 0
+        self.queue = None
+
+    cdef _getbuffer(self, Py_buffer *buffer, int flags):
+        buffer.buf = <char *>self.memory_ptr
```

Comment:
Provide function `void* DPPLUSM_GetPointer(DPPLSyclUSMRef )`.

Reply:
Or add comment why it is legal to cast ref to `char *`. Because the rule is that ref is a pointer to SYCL type.

Reply:
Fixed.